internal/cmdexport/cmdexport_test.go

```diff
@@ -99,14 +109,96 @@ jobs:
           - name: Run all kpt functions
             uses: docker://gongpu/kpt:latest
             with:
-                args: fn run . --fn-path functions/
+                args: fn run . --fn-path function.yaml
 `,
 	},
+	{
+		description: "exports a Cloud Build pipeline with --fn-path",
+		files: map[string]string{
+			"functions/function.yaml": "",
+		},
+		params: []string{
+			"cloud-build",
+			".",
+			"--fn-path",
+			"functions/",
+			"--output",
+			"cloudbuild.yaml",
+		},
+		expected: `
+steps:
+  - name: gongpu/kpt:latest
+    args:
+      - fn
+      - run
+      - .
+      - --fn-path
+      - functions/
+`,
+	},
+	{
+		description: "fails to export a Cloud Build pipeline with non-exist function path",
+		params: []string{
+			"cloud-build",
+			".",
+			"--fn-path",
+			"functions.yaml",
+			"--output",
+			"cloudbuild.yaml",
+		},
+		err: "function path (functions.yaml) does not exist",
+	},
+	{
+		description: "fails to export a Cloud Build pipeline with outside function path",
+		params: []string{
+			"cloud-build",
+			".",
+			"--fn-path",
+			"../functions/functions.yaml",
+			"--output",
+			"cloudbuild.yaml",
+		},
+		err: "function path (../functions/functions.yaml) is not within the current working directory",
+	},
+}
+
+func setupTempDir(files files) (dir string, err error) {
+	tempDir, err := ioutil.TempDir("", "kpt-fn-export-test")
+	if err != nil {
+		return "", err
+	}
+
+	for p, content := range files {
+		p = filepath.Join(tempDir, p)
+
+		err = os.MkdirAll(
+			filepath.Dir(p),
+			0755, // drwxr-xr-x
+		)
+		if err != nil {
+			return "", nil
+		}
+
+		err = ioutil.WriteFile(
+			p,
+			[]byte(content),
+			0644, // -rw-r--r--
+		)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return tempDir, nil
 }
 
 func TestCmdExport(t *testing.T) {
 	for i := range testCases {
 		testCase := testCases[i]
+		tempDir, err := setupTempDir(testCase.files)
```

Comment:
You can use defer to remove the tempDir
```
defer os.RemoveAll(tempDir)
```

Reply:
IDE prompts `Possible resource leak, 'defer' is called in a for loop.